sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -741,4 +741,20 @@ private[spark] object QueryCompilationErrors {
     new InvalidUDFClassException(s"No handler for UDAF '$name'. " +
       "Use sparkSession.udf.register(...) instead.")
   }
+
+  def unexpectedColumnExpressionError(column: Expression): Throwable = {
+    new AnalysisException(s"[BUG] unexpected column expression: $column")
+  }
+
+  def databaseFromV1SessionCatalogNotSpecifiedError(): Throwable = {
+    new AnalysisException("Database from v1 session catalog is not specified")
+  }
+
+  def nestedDatabaseUnsupportedByV1SessionCatalogError(catalog: String): Throwable = {
+    new AnalysisException(s"Nested databases are not supported by v1 session catalog: $catalog")
+  }
+
+  def invalidCallFunctionOnUnresolvedObjectError(function: String): Throwable = {
+    new UnresolvedException(s"Invalid call to $function on unresolved object")
```

Comment:
Here we can directly use `AnalysisException` and remove `UnresolvedException`.

Reply:
cc @cloud-fan . How about the suggestion?

Reply:
@allisonwang-db Please look the discussion https://github.com/apache/spark/pull/31316#discussion_r577734795

Reply:
Thanks. This makes sense to me.